### PR TITLE
3288 add grouped sciences filter option

### PR DESCRIPTION
**Pull request by richardpattinson**
### Context

Add allocation subject for grouped sciences so that it appears as a filter option

### CODE REVIEW

**Comment by richardpattinson** [comment]
[Dev: include a generic sciences option in the subject filters](https://trello.com/c/IWTnQu0R/3288-dev-include-a-generic-sciences-option-in-the-subject-filters)

File: app/helpers/course_details_helper.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@ def course_subjects
   end
 
   def all_subjects
-    @all_subjects ||= (SubjectSpecialism.pluck(:name) + AllocationSubject.pluck(:name))
+    @all_subjects ||= (SubjectSpecialism.pluck(:name) + AllocationSubject.pluck(:name) + ["Sciences - biology, chemistry, physics"])
```
Comment (mooktakim) [inline app/helpers/course_details_helper.rb:71]: You should use `Filter::ALL_SCIENCES_FILTER` here

**Comment by edwardhorsford** [comment]
> Add allocation subject for grouped sciences so that it appears as a filter option

@richardpattinson just to double check - you've not *actually* added a new allocation subject for this, right? It's purely a special case for use in the filters.

**Review by mooktakim — APPROVED** [review]
Looks good

**Comment by richardpattinson** [comment]
> > Add allocation subject for grouped sciences so that it appears as a filter option
> 
> @richardpattinson just to double check - you've not *actually* added a new allocation subject for this, right? It's purely a special case for use in the filters.

Yeah it's just an exception not a new allocation subject, my first approach was a bad take. Won't show up in any exports or similar

**Pull request merged**